marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getOpticPlan.sjs

```diff
@@ -85,7 +86,7 @@ queryOptions = fn.head(xdmp.unquote(queryOptions)).root;
  * Also replacing hyphen with underscore for column names (entity property names), schema names and view names since TDE's do the same.
  */
 const simplePropertySet = filterObjectAndArrayTypeProperties(schemaName);
-columns = columns.filter(column => simplePropertySet.has(column)).map(column => replaceHyphenWithUnderscore(column));
+columns = columns.toArray().filter(column => simplePropertySet.has(column)).map(column => replaceHyphenWithUnderscore(column));
```

Comment:
Are there existing tests ensuring that properties with hyphens have them converted into columns? Makes me think the reference project should have a new property added to it that has a hyphen in the name.